Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/PresenceTracingRiskRepository.kt

```diff
@@ -0,0 +1,118 @@
+package de.rki.coronawarnapp.presencetracing.risk
+
+import androidx.room.Dao
+import androidx.room.Entity
+import androidx.room.ForeignKey
+import androidx.room.Insert
+import androidx.room.PrimaryKey
+import androidx.room.Query
+import de.rki.coronawarnapp.eventregistration.storage.TraceLocationDatabase
+import de.rki.coronawarnapp.eventregistration.storage.entity.TraceLocationCheckInEntity
+import de.rki.coronawarnapp.risk.TraceLocationCheckInRisk
+import de.rki.coronawarnapp.util.TimeStamper
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import org.joda.time.Days
+import org.joda.time.Instant
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class PresenceTracingRiskRepository @Inject constructor(
```

Comment:
This calculates the risk when we subscribe to a flow, shouldn't we calculate the risk as part of a task and store the the tasks result and expose that as a flow? I'd have imagined this to be similar to or re-using the current `RiskLevelTask`?

Reply:
leaving it as it is for now as discussed